packages/library-base/src/commonMain/kotlin/io/realm/notifications/ListChange.kt

```diff
@@ -0,0 +1,77 @@
+package io.realm.notifications
+
+/**
+ * This sealed interface describe the possible changes that can happen to a list collection,
+ * currently [io.realm.RealmList] or [io.realm.RealmResults].
+ *
+ * The specific states are represented by the specific subclasses [InitialList], [UpdatedList] and
+ * [DeletedList].
+ *
+ * Changes can thus be consumed in a number of ways:
+ *
+ * ```
+ * // Variant 1: Switch on the sealed interface
+ * realm.filter<Person>().asFlow()
+ *   .collect { it: ListChange<RealmResults<Person>> ->
+ *       when(result) {
+ *          is InitialList -> setUIList(it.list)
+ *          is UpdatedList -> updateUIList(it) // Android RecyclerView knows how to animate ranges
+ *          is DeletedList -> deleteUIList(it.list)
+ *       }
+ *   }
+ *
+ *
+ * // Variant 2: Just pass on the list
+ * realm.filter<Person>().asFlow()
+ *   .collect { it: ListChange<RealmResults<Person>> ->
+ *       handleChange(it.list)
+ *   }
+ * ```
+ *
+ * For state of update changes, extra information is provided describing the changes from the previous
+ * version. This information is formatted in a way that can be feed directly to drive animations on UI
+ * components like `RecyclerView`. In order to access this information, the [ListChange] must be cast
+ * to the appropriate subclass.
+ *
+ * ```
+ * realm.filter<Person>().asFlow()
+ *   .collect { it: ListChange<RealmResults<Person>> ->
+ *       when(result) {
+ *          is InitialList -> setList(it.list)
+ *          is UpdatedList -> { // Automatic cast to UpdatedList
+ *              updateList(
+ *                  it.list,
+ *                  it.deletionRanges,
+ *                  it.insertionRanges,
+ *                  it.changeRanges
+ *             )
+ *          }
+ *          is DeletedList -> deleteList(it.list)
+ *       }
+ *   }
+ * ```
```

Comment:
We should probably also specify that when the list is deleted an empty list emitted instead of `null`